packages/salesforcedx-utils-vscode/src/auth/authUtil.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Aliases } from '@salesforce/core';
+import { ConfigUtil } from '../config';
+import {
+  DEFAULT_USERNAME_KEY
+} from '../constants';
+import { TelemetryService } from '../telemetry';
+
+export class AuthUtil {
+  private static instance?: AuthUtil;
+
+  public static getInstance() {
+    if (AuthUtil.instance === undefined) {
+      AuthUtil.instance = new AuthUtil();
+    }
+    return AuthUtil.instance;
+  }
+
+  public async getDefaultUsernameOrAlias(
+    enableWarning: boolean
+  ): Promise<string | undefined> {
+    try {
+      const defaultUserName = await ConfigUtil.getConfigValue(
+        DEFAULT_USERNAME_KEY
+      );
+      if (defaultUserName === undefined) {
+        return undefined;
+      }
+
+      return JSON.stringify(defaultUserName).replace(/\"/g, '');
+    } catch (err) {
+      console.error(err);
+      await TelemetryService.getInstance().sendException('get_default_username_alias', err.message);
```

Comment:
I remember running into issues when using the telemetry service in other code in the utils module where it wouldn't get initialized for the correct extension since the utils code ran before the extension initializes. Did you see something like that ?

Reply:
I'll have to think about this. At the moment, this code is only called by WorkspaceContextUtil code that would only be called after WorkspaceContextUtil has been initialized to an extension context. So the extension should be initialized. But this code could be invoked in different ways in the future. We could test for telemetry activation (`if (TelemetryService.getInstance().getReporter())`) or allow the exception to go uncaught and catch/handle telemetry at a higher level. 

Note: In looking at the above problem, I noticed that the telemetry service suffers from the same singleton problem as the channel service had (fixed by changing from a singleton channel service to a singleton-per-channel channel service), so whichever extension initializes first will get all the telemetry assigned to it. This should be changed to a singleton-per-extension telemetry service.

Reply:
FWIW it turns out the reporter check is done in the telemetry service itself. So we won't get exceptions in any case.

Reply:
very cool 👍